src/main/java/com/cdancy/jenkins/rest/features/QueueItemApi.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.cdancy.jenkins.rest.features;
+
+import java.util.List;
+
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.MediaType;
+
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.SelectJson;
+
+import com.cdancy.jenkins.rest.domain.queue.QueueItem;
+import com.cdancy.jenkins.rest.filters.JenkinsAuthenticationFilter;
+
+@RequestFilters(JenkinsAuthenticationFilter.class)
+@Consumes(MediaType.APPLICATION_JSON)
+@Path("/queue/item/{queueId}/api/json")
+public interface QueueItemApi {
+
+   @Named("queueItem:queueItem")
+   @SelectJson("")
```

Comment:
@cdancy, What do I set the `@SelectJson` argument value to, to return the queue item itself with all its fields? The json does not have a "top" name for individual items.

Reply:
You can remove it altogether so long as the returnes Object matches the main returned from the server.

Reply:
I tried removing it, but it is the same as setting to an empty string: the queue item is a null. I am stuck but I am not giving up just yet.

Reply:
I think my groovy grapes was not re-fetching from my latest local maven repo.